lib/repeatable/expression/base.rb

```diff
@@ -24,6 +24,29 @@ def to_h
         )
       end
 
+      def union(other)
+        if other.is_a?(Union)
```

Comment:
The specific use case is minimizing the persistence to JSON, and the layers of call-chains when building deeply nested rules. But as you said, it does come with some added complexity.

It might be better suited for some kind of "utilities" class to be opted into? I'll rip it out for now, and leave the commit in place so we can build it later if we ever actually need it.